libdnf/utils/locker.cpp

```diff
@@ -41,21 +43,32 @@ Locker::LockResult Locker::lock() {
     fl.l_pid = 0;
     auto rc = fcntl(lock_fd, F_SETLK, &fl);
     if (rc == -1) {
-        return LockResult::ERROR_LOCK;
+        if (errno == EACCES || errno == EAGAIN) {
+            return LockResult::ALREADY_LOCKED;
+        } else {
+            throw SystemError(errno, "Failed to obtain lock {}", path);
+        }
     }
 
     return LockResult::SUCCESS;
 }
 
 void Locker::unlock() {
     if (lock_fd != -1) {
-        unlink(path.c_str());
-        close(lock_fd);
+        if (unlink(path.c_str()) == -1) {
+            throw SystemError(errno, "Failed to unlink lock file {}", path);
+        }
+        if (close(lock_fd) == -1) {
+            throw SystemError(errno, "Failed to close lock file descriptor {}", lock_fd);
```

Comment:
I'd put the file path in the message, the file descriptor won't be useful at all, since by the time the message is displayed the process would have terminated?